File: backend/src/main/java/com/smartcampus/backend/dto/search/ItineraryRequestDTO.java
package com.smartcampus.backend.dto.search;

import java.util.List;
import lombok.*;

@Getter
@Setter
public class ItineraryRequestDTO {

    private SearchResultDTO origin;
    private SearchResultDTO destination;

    private List<String> modes;          
    private String date;                
    private String time;                 
    private Boolean arriveBy;            
    private Double walkSpeed;            
    private Double bikeSpeed;            
    private Boolean wheelchair;         

    public ItineraryRequestDTO() {
    }

}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/search/ItineraryRequestDto.kt
package com.ObservatoireCampus.mobile.model.search

data class ItineraryRequestDto(
    val origin: SearchResultDto,
    val destination: SearchResultDto,
    val modes: List<String>? = null,
    val date: String? = null,
    val time: String? = null,
    val arriveBy: Boolean? = null,
    val walkSpeed: Double? = null,
    val bikeSpeed: Double? = null,
    val wheelchair: Boolean? = null
)

fun ItineraryFilters.toRequestDto(origin: SearchResultDto, destination: SearchResultDto): ItineraryRequestDto {
    return ItineraryRequestDto(
        origin = origin,
        destination = destination,
        modes = modes.map { it.apiValue }.ifEmpty { null },
        date = date,
        time = time,
        arriveBy = arriveBy
    )
}